### PR TITLE
UX: Display tag's description as title in navigation menu

### PR DESCRIPTION
**Pull request by tgxworld**
Why this change?

We're already displaying a category's description as the title attribute
on the category section link. We should do the same for tags as well.

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/acceptance/sidebar-user-tags-section-test.js [hunk 1]
```diff
@@ -16,27 +16,6 @@ import { NotificationLevels } from "discourse/lib/notification-levels";
...
-    test("tags section is not shown", async function (assert) {
-      await visit("/");
-
-      assert.ok(
-        !exists(".sidebar-section[data-section-name='tags']"),
-        "does not display the tags section"
-      );
-    });
-  }
-);
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/sidebar-user-tags-section-test.js:38]: Moved to system tests.

File: app/assets/javascripts/discourse/tests/acceptance/sidebar-user-tags-section-test.js [hunk 2]
```diff
@@ -93,72 +72,6 @@ acceptance("Sidebar - Logged on user - Tags section", function (needs) {
...
-      tagSectionLink.textContent.trim()
-    );
-
-    assert.deepEqual(
-      tagNames,
-      ["tag1", "tag2", "tag3", "tag4"],
-      "tag section links are displayed in the right order"
-    );
-  });
-
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/sidebar-user-tags-section-test.js:161]: Moved to system tests.

File: app/serializers/site_serializer.rb [hunk 3]
```diff
@@ -250,9 +253,37 @@ def displayed_about_plugin_stat_groups
     About.displayed_plugin_stat_groups
   end
 
+  SIDEBAR_TOP_TAGS_TO_SHOW = 5
+
+  def navigation_menu_site_top_tags
+    if top_tags.present?
+      tag_names = top_tags[0...SIDEBAR_TOP_TAGS_TO_SHOW]
+      serialized = serialize_tags(Tag.where(name: tag_names))
```
Comment (tgxworld) [inline app/serializers/site_serializer.rb:261]: This is an extra query on page load but it is limited to 5 objects so it wouldn't add much overhead.

File: app/serializers/site_serializer.rb [hunk 4]
```diff
@@ -250,9 +253,37 @@ def displayed_about_plugin_stat_groups
...
+
+      # Ensures order of top tags is preserved
+      tag_names.reduce([]) do |acc, tag_name|
+        tag = serialized.find { |t| t[:name] == tag_name }
+        acc.push(tag)
+        acc
+      end
```
Comment (lis2) [inline app/serializers/site_serializer.rb:268]:
```suggestion
serialized.sort_by {|tag| tag_names.index(tag[:name]) }
```

Comment (tgxworld) [reply app/serializers/site_serializer.rb:268]: O nice :+1:

File: app/serializers/concerns/navigation_menu_tags_mixin.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module NavigationMenuTagsMixin
+  def serialize_tags(tags)
+    topic_count_column = Tag.topic_count_column(scope)
+
+    tags
+      .select(:name, topic_count_column, :pm_topic_count, :description)
+      .order(topic_count_column => :desc)
```
Comment (lis2) [inline app/serializers/concerns/navigation_menu_tags_mixin.rb:9]:
```suggestion
      .order(topic_count_column: :desc)
```

**Review by lis2 — APPROVED** [review]

File: app/serializers/concerns/navigation_menu_tags_mixin.rb (hunk 5 above)
Comment (tgxworld) [reply app/serializers/concerns/navigation_menu_tags_mixin.rb:9]: I don't think is the same. `topic_count_column` is a variable storing a symbol that may not be `:topic_count_column`.

**Pull request merged**